src/sema/unused_variable.rs

```diff
@@ -347,20 +330,31 @@ pub fn emit_warning_local_variable(variable: &symtable::Variable) -> Option<Diag
 
         VariableUsage::TryCatchErrorString => {
             if !variable.read {
-                return Some(generate_unused_warning(
+                return Some(Diagnostic::warning(
                     variable.id.loc,
-                    &format!(
+                    format!(
                         "try-catch error string '{}' has never been used",
                         variable.id.name
                     ),
-                    vec![],
                 ));
             }
 
             None
         }
         VariableUsage::AssemblyLocalVariable => {
-            //unimplemented!("Variable usage not implemented for YUL")
+            let has_value = variable.assigned || variable.initializer.has_initializer();
+            if !variable.read && !has_value {
+                return Some(Diagnostic::warning(
+                    variable.id.loc,
+                    "assembly variable has never been read or assigned".to_string(),
```

Comment:
Please do include the variable name. This is consistent with the other unused messages, and also I think explicitly repeating the variable name is useful. In an ide for example you see the diagnostics in a separate window. Calling the variable out there is good.

Also note this is what gcc/clang do as well.